File: js/curseNames.js
var nm1 = ["Acrimony","Agony","Amber","Anger","Anguish","Annihilation","Anxiety","Ash","Asperity","Assault","Atonement","Bane","Blood","Bone","Bones","Brass","Catastrophe","Chains","Chaos","Coal","Crimson","Crystal","Darkness","Dawn","Death","Defeat","Depletion","Desire","Desolation","Despair","Destruction","Doom","Dusk","Dust","Earth","Ember","Exctinction","Failure","Fire","Flame","Flesh","Flies","Frenzy","Frost","Furor","Fury","Glass","Gold","Grief","Growth","Hate","Hatred","Heartache","Horns","Hysteria","Ice","Imbalance","Ire","Iron","Isolation","Lava","Lead","Light","Loss","Love","Mania","Marble","Metal","Misery","Misfortune","Nightmares","Obsidian","Onyx","Pain","Passion","Penance","Perdition","Pests","Poison","Poverty","Pride","Prison","Rage","Rain","Rats","Ruin","Runes","Sacrifice","Sand","Scales","Scents","Seclusion","Shadows","Silence","Silk","Silver","Smiles","Smoke","Snow","Solitude","Sorrow","Spasms","Stone","Storms","Teeth","Thunder","Truth","Twilight","Umbrage","Vengeance","Venom","Vermin","Woe","Wrath","the Arachnid","the Banshee","the Bat","the Blaze","the Boulder","the Cave","the Crow","the Crown","the Crypt","the Dark","the Dead","the Dragon","the Feast","the Fire","the Flame","the Flock","the Fog","the Forge","the Grave","the Griffin","the Hallowed","the Heart","the Horn","the Hydra","the Lone Wolf","the Lonely","the Mind","the Mist","the Moon","the Nightmare","the Phoenix","the Prison","the Quill","the Rat","the Reaper","the Ring","the Ruins","the Sanguine","the Seed","the Serpent","the Shade","the Shadows","the Snail","the Storm","the Stranger","the Sun","the Talon","the Titan","the Tomb","the Tyrant","the Veil","the Vermin","the Vessel","the Visitor","the Voice","the Void","the Wild","the Wraith","the Wreckage","the Wyvern","the Zephyr"];
var nm2 = ["Acrimony","Aggression","Agony","Ancient","Angel","Anguish","Anxiety","Arachnid","Avian","Banshee","Bat","Blaze","Boulder","Brass","Canine","Chaos","Cold","Crazy","Creeper","Creeping","Crimson","Crippling","Crow","Crown","Crumbling","Crying","Crypt","Crystal","Dark","Deathbell","Delirium","Demon","Desolation","Devil","Dragon","Dread","Elephant","Ember","Ethereal","Fickle","Flock","Flying","Fog","Forbidden","Forest","Forge","Frenzy","Frost","Frozen","Furor","Fury","Ghast","Ghastly","Ghost","Grave","Harpy","Heartache","Hell","Hellish","Hopeless","Horn","Horror","Hydra","Hyper","Hysteria","Ice","Immobile","Impossible","Ire","Iron","Ironbark","Laughing","Lifeless","Limp","Limp Limb","Mania","Mind","Misery","Misfortune","Moon","Mortal","Necro","Nercrotic","Nightmare","Numb","Paralyzing","Penance","Perdition","Phantom","Pharaoh","Pygmy","Rabid","Restless","Rodent","Rotting","Scale","Seclusion","Serpent","Shadow","Shaking","Shivering","Silence","Silent","Sinister","Smiling","Spirit","Stiffening","Stoneskin","Storm","Strange","Stranger","Sun","Swamp","Terror","Thorn","Thunder","Tomb","Twilight","Undead","Vengeance","Venom","Volatile","Wandering","Wicked","Wraith","Zombie"];
var nm3 = ["Curse","Bane","Jinx","Hex","Vex","Cure","Curse","Curse","Curse","Curse","Curse"];

var br = "";

function nameGen(){
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd2 = Math.floor(Math.random() * nm3.length);
		if(i < 5){
			rnd = Math.floor(Math.random() * nm1.length);
			names = nm3[rnd2] + " of " + nm1[rnd];
		}else{
			rnd = Math.floor(Math.random() * nm2.length);
			while(nm3[rnd2] === nm2[rnd]){
				rnd = Math.floor(Math.random() * nm2.length);
			}
			names = "The " + nm2[rnd] + " " + nm3[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/curseNames.py
__all__ = ['curseNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nm3', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
            if (var.get('i')<Js(5.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('names', ((var.get('nm3').get(var.get('rnd2'))+Js(' of '))+var.get('nm1').get(var.get('rnd'))))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                while PyJsStrictEq(var.get('nm3').get(var.get('rnd2')),var.get('nm2').get(var.get('rnd'))):
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', (((Js('The ')+var.get('nm2').get(var.get('rnd')))+Js(' '))+var.get('nm3').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Acrimony'), Js('Agony'), Js('Amber'), Js('Anger'), Js('Anguish'), Js('Annihilation'), Js('Anxiety'), Js('Ash'), Js('Asperity'), Js('Assault'), Js('Atonement'), Js('Bane'), Js('Blood'), Js('Bone'), Js('Bones'), Js('Brass'), Js('Catastrophe'), Js('Chains'), Js('Chaos'), Js('Coal'), Js('Crimson'), Js('Crystal'), Js('Darkness'), Js('Dawn'), Js('Death'), Js('Defeat'), Js('Depletion'), Js('Desire'), Js('Desolation'), Js('Despair'), Js('Destruction'), Js('Doom'), Js('Dusk'), Js('Dust'), Js('Earth'), Js('Ember'), Js('Exctinction'), Js('Failure'), Js('Fire'), Js('Flame'), Js('Flesh'), Js('Flies'), Js('Frenzy'), Js('Frost'), Js('Furor'), Js('Fury'), Js('Glass'), Js('Gold'), Js('Grief'), Js('Growth'), Js('Hate'), Js('Hatred'), Js('Heartache'), Js('Horns'), Js('Hysteria'), Js('Ice'), Js('Imbalance'), Js('Ire'), Js('Iron'), Js('Isolation'), Js('Lava'), Js('Lead'), Js('Light'), Js('Loss'), Js('Love'), Js('Mania'), Js('Marble'), Js('Metal'), Js('Misery'), Js('Misfortune'), Js('Nightmares'), Js('Obsidian'), Js('Onyx'), Js('Pain'), Js('Passion'), Js('Penance'), Js('Perdition'), Js('Pests'), Js('Poison'), Js('Poverty'), Js('Pride'), Js('Prison'), Js('Rage'), Js('Rain'), Js('Rats'), Js('Ruin'), Js('Runes'), Js('Sacrifice'), Js('Sand'), Js('Scales'), Js('Scents'), Js('Seclusion'), Js('Shadows'), Js('Silence'), Js('Silk'), Js('Silver'), Js('Smiles'), Js('Smoke'), Js('Snow'), Js('Solitude'), Js('Sorrow'), Js('Spasms'), Js('Stone'), Js('Storms'), Js('Teeth'), Js('Thunder'), Js('Truth'), Js('Twilight'), Js('Umbrage'), Js('Vengeance'), Js('Venom'), Js('Vermin'), Js('Woe'), Js('Wrath'), Js('the Arachnid'), Js('the Banshee'), Js('the Bat'), Js('the Blaze'), Js('the Boulder'), Js('the Cave'), Js('the Crow'), Js('the Crown'), Js('the Crypt'), Js('the Dark'), Js('the Dead'), Js('the Dragon'), Js('the Feast'), Js('the Fire'), Js('the Flame'), Js('the Flock'), Js('the Fog'), Js('the Forge'), Js('the Grave'), Js('the Griffin'), Js('the Hallowed'), Js('the Heart'), Js('the Horn'), Js('the Hydra'), Js('the Lone Wolf'), Js('the Lonely'), Js('the Mind'), Js('the Mist'), Js('the Moon'), Js('the Nightmare'), Js('the Phoenix'), Js('the Prison'), Js('the Quill'), Js('the Rat'), Js('the Reaper'), Js('the Ring'), Js('the Ruins'), Js('the Sanguine'), Js('the Seed'), Js('the Serpent'), Js('the Shade'), Js('the Shadows'), Js('the Snail'), Js('the Storm'), Js('the Stranger'), Js('the Sun'), Js('the Talon'), Js('the Titan'), Js('the Tomb'), Js('the Tyrant'), Js('the Veil'), Js('the Vermin'), Js('the Vessel'), Js('the Visitor'), Js('the Voice'), Js('the Void'), Js('the Wild'), Js('the Wraith'), Js('the Wreckage'), Js('the Wyvern'), Js('the Zephyr')]))
var.put('nm2', Js([Js('Acrimony'), Js('Aggression'), Js('Agony'), Js('Ancient'), Js('Angel'), Js('Anguish'), Js('Anxiety'), Js('Arachnid'), Js('Avian'), Js('Banshee'), Js('Bat'), Js('Blaze'), Js('Boulder'), Js('Brass'), Js('Canine'), Js('Chaos'), Js('Cold'), Js('Crazy'), Js('Creeper'), Js('Creeping'), Js('Crimson'), Js('Crippling'), Js('Crow'), Js('Crown'), Js('Crumbling'), Js('Crying'), Js('Crypt'), Js('Crystal'), Js('Dark'), Js('Deathbell'), Js('Delirium'), Js('Demon'), Js('Desolation'), Js('Devil'), Js('Dragon'), Js('Dread'), Js('Elephant'), Js('Ember'), Js('Ethereal'), Js('Fickle'), Js('Flock'), Js('Flying'), Js('Fog'), Js('Forbidden'), Js('Forest'), Js('Forge'), Js('Frenzy'), Js('Frost'), Js('Frozen'), Js('Furor'), Js('Fury'), Js('Ghast'), Js('Ghastly'), Js('Ghost'), Js('Grave'), Js('Harpy'), Js('Heartache'), Js('Hell'), Js('Hellish'), Js('Hopeless'), Js('Horn'), Js('Horror'), Js('Hydra'), Js('Hyper'), Js('Hysteria'), Js('Ice'), Js('Immobile'), Js('Impossible'), Js('Ire'), Js('Iron'), Js('Ironbark'), Js('Laughing'), Js('Lifeless'), Js('Limp'), Js('Limp Limb'), Js('Mania'), Js('Mind'), Js('Misery'), Js('Misfortune'), Js('Moon'), Js('Mortal'), Js('Necro'), Js('Nercrotic'), Js('Nightmare'), Js('Numb'), Js('Paralyzing'), Js('Penance'), Js('Perdition'), Js('Phantom'), Js('Pharaoh'), Js('Pygmy'), Js('Rabid'), Js('Restless'), Js('Rodent'), Js('Rotting'), Js('Scale'), Js('Seclusion'), Js('Serpent'), Js('Shadow'), Js('Shaking'), Js('Shivering'), Js('Silence'), Js('Silent'), Js('Sinister'), Js('Smiling'), Js('Spirit'), Js('Stiffening'), Js('Stoneskin'), Js('Storm'), Js('Strange'), Js('Stranger'), Js('Sun'), Js('Swamp'), Js('Terror'), Js('Thorn'), Js('Thunder'), Js('Tomb'), Js('Twilight'), Js('Undead'), Js('Vengeance'), Js('Venom'), Js('Volatile'), Js('Wandering'), Js('Wicked'), Js('Wraith'), Js('Zombie')]))
var.put('nm3', Js([Js('Curse'), Js('Bane'), Js('Jinx'), Js('Hex'), Js('Vex'), Js('Cure'), Js('Curse'), Js('Curse'), Js('Curse'), Js('Curse'), Js('Curse')]))
pass
pass


# Add lib to the module scope
curseNames = var.to_python()